File: testcases/src/main/java/ir/ayantech/testcases/TestcaseListActivity.java
package ir.ayantech.testcases;

import android.graphics.Typeface;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;

import ir.ayantech.versioncontrol.VersionControlCore;

public class TestcaseListActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_testcase_list);
        bindButtons();
    }

    private void bindButtons() {
        findViewById(R.id.testcase1Button).setOnClickListener(this);
        findViewById(R.id.testcase2Button).setOnClickListener(this);
        findViewById(R.id.testcase3Button).setOnClickListener(this);
        findViewById(R.id.testcase4Button).setOnClickListener(this);
        findViewById(R.id.testcase5Button).setOnClickListener(this);

        findViewById(R.id.shareBtn).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                VersionControlCore.getInstance().shareApp(TestcaseListActivity.this);
            }
        });
    }

    @Override
    public void onClick(View v) {
        Integer testcaseNumber = getTestcaseNumber(v);
        String testcaseName = getTestcaseName(testcaseNumber);
        runVersionControlTestcase(testcaseName);
    }

    @NonNull
    private Integer getTestcaseNumber(View v) {
        int testcaseNumber = 0;
        switch (v.getId()) {
            case R.id.testcase1Button:
                testcaseNumber = 1;
                break;

            case R.id.testcase2Button:
                testcaseNumber = 2;
                break;

            case R.id.testcase3Button:
                testcaseNumber = 3;
                break;

            case R.id.testcase4Button:
                testcaseNumber = 4;
                break;

            case R.id.testcase5Button:
                testcaseNumber = 5;
                break;
        }
        return testcaseNumber;
    }

    @NonNull
    private String getTestcaseName(Integer testcaseNumber) {
        return "testcase" + testcaseNumber;
    }

    private void runVersionControlTestcase(String testcaseName) {
        VersionControlCore.getInstance()
                .setApplicationName(testcaseName)
                .setCategoryName("cafebazar")
                .setTypeface(Typeface.createFromAsset(getAssets(), "fonts/iransans-ultralight.ttf"))
                .checkForNewVersion(this);
    }
}

File: testcases/src/main/java/ir/ayantech/testcases/TestcaseListActivity.kt
package ir.ayantech.testcases

import android.graphics.Typeface
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import ir.ayantech.versioncontrol.VersionControlCore

class TestcaseListActivity : AppCompatActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_testcase_list)
        bindButtons()
    }

    private fun bindButtons() {
        findViewById<View>(R.id.testcase1Button).setOnClickListener(this)
        findViewById<View>(R.id.testcase2Button).setOnClickListener(this)
        findViewById<View>(R.id.testcase3Button).setOnClickListener(this)
        findViewById<View>(R.id.testcase4Button).setOnClickListener(this)
        findViewById<View>(R.id.testcase5Button).setOnClickListener(this)

        findViewById<View>(R.id.shareBtn).setOnClickListener {
            VersionControlCore.getInstance("https://versioncontrol.infra.ayantech.ir/WebServices/App.svc/").shareApp(this@TestcaseListActivity)
        }
    }

    override fun onClick(v: View) {
        val testcaseNumber = getTestcaseNumber(v)
        val testcaseName = getTestcaseName(testcaseNumber)
        runVersionControlTestcase(testcaseName)
    }

    private fun getTestcaseNumber(v: View): Int {
        return when (v.id) {
            R.id.testcase1Button -> 1
            R.id.testcase2Button -> 2
            R.id.testcase3Button -> 3
            R.id.testcase4Button -> 4
            R.id.testcase5Button -> 5
            else -> 0
        }
    }

    private fun getTestcaseName(testcaseNumber: Int): String {
        return "testcase$testcaseNumber"
    }

    private fun runVersionControlTestcase(testcaseName: String) {
        VersionControlCore.getInstance("https://versioncontrol.infra.ayantech.ir/WebServices/App.svc/")
            .setApplicationName(testcaseName)
            .setCategoryName("cafebazaar")
            .setTypeface(Typeface.createFromAsset(assets, "fonts/iransans-ultralight.ttf"))
            .checkForNewVersion(this)
    }
}
